### DISCUSSION TITLE
HTTP/2 Response for 404 scenario

### DISCUSSION THREAD

**Discussion by AmandeepSingh97** [netty/netty#11783] [Q&A]
Hello All,
We have observed that Netty has a different response structure for HTTP/2 working calls (status 200) and not found (400 status).

For status 200:
- First we receive DefaultHttpResponse
- Second we receive DefaultLastHttpContent

For status 404:
- We receive DefaultFullHttpResponse (which is kind of like a single version for both DefaultHttpResponse and DefaultLastHttpContent)
I assume it is done to save 1 transfer ? Can someone please throw some clarity on this?
Is there a major reason for this difference in behaviour apart from the one I mentioned above?

Netty version used: 4.1.69.FINAL

**Comment by NiteshKant** [comment]
Can you share how you configure your pipeline?

**Comment by hyperxpro** [comment]
Also, please share code snippets so it's easier to understand how requests and responses are being generated.
